### PR TITLE
Add support for non-free MapQuest API

### PR DESCRIPTION
**Pull request by pratheekrebala**
Small addition to create support for the non-free MapQuest API. It's mostly copied from the Mapbox provider with slight changes to fit the Mapquest API (documented [here](https://developer.mapquest.com/documentation/geocoding-api/))

### CODE REVIEW

File: geopy/geocoders/__init__.py [hunk 1]
```diff
@@ -169,6 +170,7 @@
     "tomtom": TomTom,
     "what3words": What3Words,
     "yandex": Yandex,
+    "mapquest": MapQuest
```
Comment (KostyaEsmukov) [inline geopy/geocoders/__init__.py:173]: 1. The list should be sorted
2. Please take a look at the comment on the 81st line: it's not the only place where a new geocoder should be added.

File: geopy/geocoders/mapquest.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        timeout=DEFAULT_SENTINEL,
+        proxies=DEFAULT_SENTINEL,
+        user_agent=None,
+        ssl_context=DEFAULT_SENTINEL,
+        domain='www.mapquestapi.com',
+    ):
+        """
+        :param str api_key: The API key required by Mapbox to perform
+            geocoding requests. API keys are managed through Mapox's account
+            page (https://www.mapbox.com/account/access-tokens).
```
Comment (KostyaEsmukov) [inline geopy/geocoders/mapquest.py:32]: Shouldn't that be MapQuest instead of MapBox?

**Comment by KostyaEsmukov** [comment]
> Small addition to create support for the non-free MapQuest API. It's mostly copied from the Mapbox provider with slight changes to fit the Mapquest API (documented [here](http://developers.google.com/maps/documentation/geocoding))

This sentence is very confusing:
1. Does MapQuest use Mapbox-alike API (like OpenMapQuest uses Nominatim)? Or did you mean that you just chose MapBox as a skeleton for creating a new geocoding class (completely unrelated to MapBox)?
2. The "here" documentation link leads to Google, looks like it should've been https://developer.mapquest.com/documentation/geocoding-api/

**Comment by pratheekrebala** [comment]
> > Small addition to create support for the non-free MapQuest API. It's mostly copied from the Mapbox provider with slight changes to fit the Mapquest API (documented [here](http://developers.google.com/maps/documentation/geocoding))
> 
> This sentence is very confusing:
> 
>    1. Does MapQuest use Mapbox-alike API (like OpenMapQuest uses Nominatim)? Or did you mean that you just chose MapBox as a skeleton for creating a new geocoding class (completely unrelated to MapBox)?

Sorry about the confusion. Your interpretation is correct: I used the MapBox as a skeleton to generate the Mapquest API.

>    2. The "here" documentation link leads to Google, looks like it should've been https://developer.mapquest.com/documentation/geocoding-api/

That's also the correct link to the documentation. I've updated my initial PR to point to the correct link.

File: geopy/geocoders/mapquest.py (hunk 2 above)
Comment (pratheekrebala) [reply geopy/geocoders/mapquest.py:32]: This is correct. I will make another commit this evening to change all these references. Apologies about the sight.

File: geopy/geocoders/__init__.py (hunk 1 above)
Comment (pratheekrebala) [reply geopy/geocoders/__init__.py]: I think I added the geocoder in all the places. Could you confirm?

File: geopy/geocoders/mapquest.py (hunk 2 above)
Comment (pratheekrebala) [reply geopy/geocoders/mapquest.py]: All the references to MapBox have been changed to MapQuest.

**Comment by pratheekrebala** [comment]
I think I have fixed all the changes you have requested and added tests to `test/geocoders/mapquest.py`. I ensured that the tests were passing on my local machine using my MapQuest key.

Let me know if you need a key to add to the CI.

Thank you for taking the time to review this PR!

**Pull request merged**